src/Features/LanguageServer/Protocol/Features/Options/SolutionCrawlerOptionsStorage.cs

```diff
@@ -48,4 +56,36 @@ public static BackgroundAnalysisScope GetBackgroundAnalysisScope(this IGlobalOpt
         return globalOptions.GetOption(SolutionBackgroundAnalysisScopeOption) ??
                globalOptions.GetOption(BackgroundAnalysisScopeOption, language);
     }
+
+    public static bool IsFullSolutionAnalysisEnabled(this DiagnosticAnalyzer analyzer, IGlobalOptionService globalOptions, string language)
+    {
+        if (analyzer.IsCompilerAnalyzer())
+        {
+            return globalOptions.GetOption(CompilerDiagnosticsScopeOption, language) == CompilerDiagnosticsScope.FullSolution;
+        }
+
+        return GetBackgroundAnalysisScope(globalOptions, language) == BackgroundAnalysisScope.FullSolution;
+    }
+
+    public static bool IsFullSolutionAnalysisEnabled(
```

Comment:
tbh, it feels a bit weird to have this helper, but hten have the callers have to introspect the two subvalues.  i think i'd prefer to just get the two values, and then process the different combinations of them at the poitn that this helper is called.

Reply:
> tbh, it feels a bit weird to have this helper, but hten have the callers have to introspect the two subvalues

Most callsites actually ignore the out params, and we don't want these to have the burden of understanding the individual values and deduce the effective value. I think only 1-2 callsites care about individual values for compiler diagnostics scope and analyzer diagnostics scope. Let me add another overload here which just returns a bool without out params and doc both methods properly.